tests/core/test_versioning.py

```
@@ -0,0 +1,289 @@
+        v1 = versioning.register_version(1, 0, 0, VersionStatus.STABLE, basic_schema)
+        v2 = versioning.register_version(1, 1, 0, VersionStatus.STABLE, basic_schema)
+        v3 = versioning.register_version(2, 0, 0, VersionStatus.BETA, basic_schema)
+
+        assert versioning._is_newer_version(v2, v1)
+        assert versioning._is_newer_version(v3, v2)
+        assert not versioning._is_newer_version(v1, v2)
+
+    def test_migration_path(self, versioning, basic_schema):
+        """Test migration path finding."""
+        v1 = versioning.register_version(1, 0, 0, VersionStatus.STABLE, basic_schema)
+        v2 = versioning.register_version(1, 1, 0, VersionStatus.STABLE, basic_schema)
```

Ruff (0.8.2)

Local variable `v2` is assigned to but never used

Remove assignment to unused variable `v2`

(F841)